app/models/loan.rb

```diff
@@ -68,6 +69,9 @@ class Loan < Project
 
   validates :organization_id, presence: true
 
+  before_create :build_loan_health_check
+  after_commit :recalculate_loan
```

Comment:
Funny story about this. I only know about `after_commit` because I have used Sidekiq in the past. Sidekiq picks up jobs so fast, that if you use after_save Sidekiq is already processing your job before your data is committed to the DB.